spark/v3.4/spark/src/jmh/java/org/apache/iceberg/spark/data/parquet/SparkParquetReadersFlatDataBenchmark.java

```diff
@@ -214,6 +216,8 @@ public void readWithProjectionUsingSparkReader(Blackhole blackhole) throws IOExc
             .set("spark.sql.parquet.binaryAsString", "false")
             .set("spark.sql.parquet.int96AsTimestamp", "false")
             .set("spark.sql.caseSensitive", "false")
+            .set("spark.sql.parquet.inferTimestampNTZ.enabled", "false")
+            .set("spark.sql.legacy.parquet.nanosAsLong", "false")
```

Comment:
Why do we need the second one? It seems it was added in 3.2.

Reply:
both flags were just recently added to https://github.com/apache/spark/blame/branch-3.4/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L72-L73

Reply:
it seems that `spark.sql.legacy.parquet.nanosAsLong` might be required for our Spark [3.3](https://github.com/apache/spark/blame/v3.2.4/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala) + [3.2](https://github.com/apache/spark/blame/v3.3.2/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L70) benchmarks as well

Reply:
I've included a fix for the Spark 3.3 benchmarks in this PR